### PR TITLE
Fixed extinguishing items on boats

### LINKED ISSUE #45827: Your joint is quenched by water (when in a boat)
# Describe the bug
When you smoke in a boat, your stoge goes out.

# Steps To Reproduce
Burn one down in a boat. It will get wet and go out.

# Expected behavior
Survivors can row out into a lake and smoke while fishing.

### PR DESCRIPTION
**Pull request by thomaswisneski**
You can smoke a joint on a boat now.

#### Summary

SUMMARY: Bugfixes "Fixed items extinguishing on boats"

#### Purpose of change
Fixes #45827. Items in boats are no longer improperly extinguished

#### Describe the solution

Added logic to extinguishing checks to check if the player/item are in a vehicle.

#### Testing

Smoked some joints in a boat. Threw some joints in the water. Dropped some joints in the boat.

### CODE REVIEW

File: src/item.cpp [hunk 1]
```diff
@@ -9727,11 +9728,11 @@ bool item::process_extinguish( player *carrier, const tripoint &pos )
             break;
     }
     map &here = get_map();
-    if( in_inv && here.has_flag( flag_DEEP_WATER, pos ) ) {
+    if( in_inv && here.has_flag( flag_DEEP_WATER, pos ) && !in_veh) {
```
Comment (mlangsdorf) [inline src/item.cpp:9731]:
```suggestion
    if( in_inv && !in_veh && here.has_flag( flag_DEEP_WATER, pos ) ) {
```
closing parenthesis should have a space before it, and the quick, pre-calculated value should be checked before the relatively expensive flag check.

File: src/item.cpp [hunk 2]
```diff
@@ -9727,11 +9728,11 @@ bool item::process_extinguish( player *carrier, const tripoint &pos )
...
         extinguish = true;
         submerged = true;
     }
-    if( ( !in_inv && here.has_flag( flag_LIQUID, pos ) ) ||
+    if( ( !in_inv && here.has_flag( flag_LIQUID, pos ) && !here.veh_at(pos) ) ||
```
Comment (mlangsdorf) [inline src/item.cpp:9735]:
```suggestion
    if( ( !in_inv && here.has_flag( flag_LIQUID, pos ) && !here.veh_at( pos ) ) ||
```
spaces go after open parenthesis and before close parenthesis.

**Review by mlangsdorf — COMMENTED** [review]
Please astyle your code properly.  Otherwise, looks good.

**Comment by thomasrwisneski** [comment]
Thanks for letting me know. I'll be more aware of the coding style from now on.

**Comment by anothersimulacrum** [comment]
If you're using make, you can run `make astyle` to style your code.
If you're using VS, there's some sort of extension.

**Comment by ZhilkinSerg** [comment]
https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/DEVELOPER_TOOLING.md#code-style-astyle

**Pull request merged**